crypto/serializer/serializer_lib.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/bio.h>
+#include <openssl/serializer.h>
+#include "serializer_local.h"
+
+int OSSL_SERIALIZER_CTX_set0_object(OSSL_SERIALIZER_CTX *ctx,
+                                    const void *object)
+{
+    if (ctx == NULL) {
+        ERR_raise(ERR_LIB_OSSL_SERIALIZER, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+
+    ctx->object = object;
+    return 1;
+}
+
+const void *OSSL_SERIALIZER_CTX_get0_object(OSSL_SERIALIZER_CTX *ctx)
+{
+    if (ctx == NULL) {
+        ERR_raise(ERR_LIB_OSSL_SERIALIZER, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+
+    return ctx->object;
+}
+
+int OSSL_SERIALIZER_CTX_set_do_output(OSSL_SERIALIZER_CTX *ctx,
+                                      int (*do_output)(OSSL_SERIALIZER_CTX *ctx,
+                                                       BIO *out))
+{
+    if (ctx == NULL) {
```

Comment:
I dislike these NULL checks.  Is that policy now?  We don't do it in most places.

Reply:
[sigh]

We don't never had an actual policy on the matter as far as I know, and depending on who you ask, you get very different answers.

Reply:
I believe we don't check most of the time and it would be good for consistency to follow that practice.  Perhaps @mspncp can come up with a "git grep" pattern to answer for sure :)

Reply:
I can make them `ossl_assert`s if that suits everyone better. *That* would be consistent if anything! 'cause if you think about it, not checking NULL is a hard assertion, and we have generally said no to hard assertions, so one might ask why we should have that specifically for pointers?

Reply:
I dont want to beat a dead horse, as they say, but OpenSSL does not defend against many types programming errors, such as passing random data into a bignum or dong memcmp with a NULL argument.

Turning programming errors, that would normally be crashes, into run-time errors is controversial at best.  Runtime errors as we have seen  are often/usually ignored.

Reply:
Considering how we go back and forth on this *as a group*, it is indeed controversial.

Reply:
I've taken up this question on [openssl-project](https://mta.openssl.org/pipermail/openssl-project/2019-November/001646.html)

Meanwhile, I don't think that question should be a show-stopper for this PR.  This check can be adjusted later on.